juntagrico/util/lifecycle.py

```diff
@@ -0,0 +1,5 @@
+def handle_activated_deactivated(instance, sender, activated, deactivated):
+    if instance._old['active'] != instance.active and instance._old['active'] is False and instance.deactivation_date is None:
+        activated.send(sender=sender, instance=instance)
+    elif instance._old['active'] != instance.active and instance._old['active'] is True and instance.deactivation_date is None:
```

Comment:
Same for `and instance.deactivation_date is None`.
Maybe 2 nested ifs are easier to read here.

Reply:
the first one checks from from inactive to active and this one checks from active to inactive. 
but the deactivation date check can be be on a higher if statement i will check that

Reply:
Isn't the deactivation_date always None, if the instance is active? Otherwise the state would be inconsistent. If yes, checking for the deactivation_date is redundant.

Reply:
it's always none beacause of this checks since it prevents reactivation. sure we check  that elswhere as well but it is good to check it here again. and if for some weird reason the active flag is true and the deactivation date is not none this code just does nothing and does not make things worse